core/src/main/java/org/bitcoinj/core/Peer.java

```diff
@@ -1313,19 +1314,19 @@ public ListenableCompletableFuture<AddressMessage> getAddr() {
      * isn't known until their headers are available and they are requested in chunks, so some headers may be downloaded
      * twice using this scheme, but this optimization can still be a large win for newly created wallets.
      *
-     * @param secondsSinceEpoch Time in seconds since the epoch or 0 to reset to always downloading block bodies.
+     * @param fastCatchupTime time before which block bodies are skipped, or empty to always download block bodies
      */
-    public void setDownloadParameters(long secondsSinceEpoch, boolean useFilteredBlocks) {
+    public void setDownloadParameters(Optional<Instant> fastCatchupTime, boolean useFilteredBlocks) {
```

Comment:
Optional<> is most useful as a return value, it is generally not recommended to use Optional<> as a parameter to a method.

In this case, since we are creating new methods and deprecating the existing method, maybe something like:

`setFastDownload(boolean useFilteredBlocks, Instant fastCatchupTime)`
and
`setNormalDownload(boolean useFilteredBlocks)`
?

Reply:
Ok, I split this up into two methods like you suggested. I feel there are already signs of boilerplate growth, but indeed in this case it might be worth it. Ready for another review.